source/validate_cfg.cpp

```diff
@@ -143,10 +149,14 @@ const vector<BasicBlock*>* successor(const BasicBlock* b) {
   return b->get_successors();
 }
 
+const vector<BasicBlock*>* predecessor(const BasicBlock* b) {
+  return b->get_predecessors();
+}
+
 }  // namespace
 
 vector<pair<BasicBlock*, BasicBlock*>> CalculateDominators(
-    vector<cbb_ptr>& postorder) {
+    vector<cbb_ptr>& postorder, get_blocks_func predecessor_func) {
```

Comment:
Would it be better if I removed these functions from the header and put them in the anonymous namespace in the current file? It is not currently being used in other functions.